libcalico-go/lib/backend/k8s/k8s.go

```diff
@@ -409,6 +415,7 @@ func (c *KubeClient) Clean() error {
 		apiv3.KindIPReservation,
 		apiv3.KindHostEndpoint,
 		apiv3.KindKubeControllersConfiguration,
+		apiv3.KindIPAMConfiguration,
```

Comment:
I think the `Kind` here needs to be the CRD kind (IPAMConfig) not the apiserver kind (IPAMConfiguration)